src/main/java/com/google/sps/servlets/AddSchoolData.java

```diff
@@ -0,0 +1,133 @@
+package com.google.sps.servlets.second;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-info")
+public class AddSchoolData extends HttpServlet {
+
+  private Boolean isNewSchool = true;
+  private Boolean isNewCourse = true;
+  private Boolean isNewProfessor = true;
+
+  private Key existingSchoolKey;
+  private Key existingCourseKey;
+  private Key existingProfessorKey;
+
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    addSchoolData(db, request);
+    response.setContentType("text/html; charset=UTF-8");
+    response.setCharacterEncoding("UTF-8");
+    response.sendRedirect("/addSchoolInfo.html");
+  }
+
+  public List<Entity> findQueryMatch(
+      DatastoreService db, String entityType, String entityProperty, String propertyValue) {
+    Filter filter = new FilterPredicate(entityProperty, FilterOperator.EQUAL, propertyValue);
+    Query q = new Query(entityType).setFilter(filter);
+    List<Entity> result = db.prepare(q).asList(FetchOptions.Builder.withDefaults());
+    return result;
+  }
+
+  public void isNewFlagSwitcher(
+      DatastoreService db, String schoolName, String courseName, String profName) {
+    List<Entity> querySchool = findQueryMatch(db, "school", "school-name", schoolName);
+    if (!querySchool.isEmpty()) {
+      existingSchoolKey = querySchool.get(0).getKey();
+      isNewSchool = false;
+    }
+
+    List<Entity> queryCourse = findQueryMatch(db, "course", "course-name", courseName);
+    if (!queryCourse.isEmpty()) {
+      existingSchoolKey = queryCourse.get(0).getKey();
+      isNewCourse = false;
+    }
+
+    List<Entity> queryProfessor = findQueryMatch(db, "professor", "professor-name", profName);
+    if (!queryProfessor.isEmpty()) {
+      existingSchoolKey = queryProfessor.get(0).getKey();
+      isNewProfessor = false;
+    }
+  }
+
+  public Entity createSchool(DatastoreService db, String schoolName) {
```

Comment:
If you initialize datastore at the beginning, do you still need to pass in db as a param?